### DISCUSSION TITLE
How can a new Coder help

### DISCUSSION THREAD

**Discussion by Arch-druid** [DeepSOIC/Lattice2#63] [General]
What language is the Lattice2 workbench coded in?
Does the Lattice 2 intend to overlap/replace functions of other workbenches? If so, then I would love to help  as many of the other features in free cad have odd implementations that haven't been visited or are simply circumvented with other benches. Would appreciate any way that I can assist in contributing, even if it's just making icons!

**Comment by DeepSOIC** [comment]
Hi! Python, of course, github straight up tells you that!

Lattice2 does replace some features of FreeCAD, and also some features of Lattice2 have been copied into FreeCAD (by me and by others), and some were made obsolete. L2 is not intended to replace FreeCAD's functionality, but rather extend it. Also, you may find similar concepts in Assembly4 addon.

If you want to just help with something - a good place is to make task dialogs for some tools, such as arrays, as editing them with property editor is quite a bit annoying. Also there are a few features begging to exist that i have still not bothered to make: path array, 2d arrays. Also preferences page - there is none, but there are a few things begging to be there.
